### PR TITLE
SASS-1795 added delete state benefits EOY end point

### PR DESCRIPTION
**Pull request by HarryGull**
Add delete state benefits end of year api endpoint

[SASS-1795](https://jira.tools.tax.service.gov.uk/browse/SASS-1795)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [ ]  Have you checked the PR Builder passes?

### CODE REVIEW

File: conf/app.routes [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
-GET     /pension-reliefs/nino/:nino/taxYear/:taxYear              controllers.PensionReliefsController.getPensionReliefs(nino: String, taxYear: Int)
+GET     /pension-reliefs/nino/:nino/taxYear/:taxYear                          controllers.PensionReliefsController.getPensionReliefs(nino: String, taxYear: Int)
 
-PUT     /pension-reliefs/nino/:nino/taxYear/:taxYear              controllers.PensionReliefsController.createOrAmendPensionReliefs(nino: String, taxYear: Int)
+PUT     /pension-reliefs/nino/:nino/taxYear/:taxYear                          controllers.PensionReliefsController.createOrAmendPensionReliefs(nino: String, taxYear: Int)
 
-DELETE  /pension-reliefs/nino/:nino/taxYear/:taxYear              controllers.PensionReliefsController.deletePensionReliefs(nino: String, taxYear: Int)
+DELETE  /pension-reliefs/nino/:nino/taxYear/:taxYear                          controllers.PensionReliefsController.deletePensionReliefs(nino: String, taxYear: Int)
+
+DELETE  /state-benefits/nino/:nino/taxYear/:taxYear/benefitId/:benefitId      controllers.StateBenefitsController.deleteStateBenefits(nino: String, taxYear: Int, benefitId: String)
```
Comment (ina-tu) [inline conf/app.routes:15]: this URL is different to the one in the DES spec

**Review by ina-tu — COMMENTED** [review]
needs a rebase

File: it/api/DeleteStateBenefitsITest.scala [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        .delete()) {
+        result =>
+          result.status mustBe NOT_FOUND
+          Json.parse(result.body) mustBe Json.obj(
+            "code" -> "NO_DATA_FOUND", "reason" -> "The remote endpoint has indicated that the requested resource could not be found.")
+      }
+    }
+
+
+    "return 503 Forbidden if the state benefit cannot be deleted" in new Setup {
```
Comment (ina-tu) [inline it/api/DeleteStateBenefitsITest.scala:97]: should be 403

File: app/models/DesErrorBody.scala [hunk 3]
```diff
@@ -43,8 +43,11 @@ object DesErrorBodyModel {
     "INVALID_CORRELATIONID", "Submission has not passed validation. Invalid Header parameter CorrelationId.")
   val invalidPayload: DesErrorBodyModel = DesErrorBodyModel(
     "INVALID_PAYLOAD", "Submission has not passed validation. Invalid payload.")
+  val invalidBenefitId: DesErrorBodyModel = DesErrorBodyModel(
+    "INVALID_BENEFIT_ID", "Submission has not passed validation. Invalid parameter benefitId.")
```
Comment (ina-tu) [inline app/models/DesErrorBody.scala:47]: you could add NO_DATA_FOUND here too

File: conf/app.routes (hunk 1 above)
Comment (HarryGull) [reply conf/app.routes]: Yes this is the URL to this service and not the URL to call DES

The matching URL in the DES spec you will find in the StateBenefitsConnector where it calls DES with that URL
(line 49)

File: it/api/DeleteStateBenefitsITest.scala (hunk 2 above)
Comment (HarryGull) [reply it/api/DeleteStateBenefitsITest.scala]: changed

File: app/models/DesErrorBody.scala (hunk 3 above)
Comment (HarryGull) [reply app/models/DesErrorBody.scala:49]: added

**Review by ina-tu — APPROVED** [review]

**Pull request merged**